crates/recursor/src/recursor.rs

```diff
@@ -484,14 +491,16 @@ enum RecursorMode {
 
 #[cfg(feature = "dnssec")]
 mod for_dnssec {
-    use std::time::Instant;
+    use std::{sync::Arc, time::Instant};
 
     use futures_util::{
         future,
         stream::{self, BoxStream},
         StreamExt as _, TryFutureExt as _,
     };
 
+    use parking_lot::Mutex;
```

Comment:
Ohh, sorry for not catching this sooner, but I think the sensible thing to do here is to use an `AtomicU8` (probably wrapped in `Arc`?) instead of `Mutex<u8>`, which should be a lot faster and more compact. I think we can probably use `Ordering::Relaxed` for this.